src/pages/audio-rewards-page/store/sagas.test.ts

```diff
@@ -0,0 +1,331 @@
+import { getAccountUser } from 'common/store/account/selectors'
+import {
+  getClaimStatus,
+  getClaimToRetry,
+  getUserChallenge
+} from 'common/store/pages/audio-rewards/selectors'
+import {
+  Claim,
+  claimChallengeReward,
+  claimChallengeRewardFailed,
+  claimChallengeRewardSucceeded,
+  claimChallengeRewardWaitForRetry,
+  ClaimStatus,
+  CognitoFlowStatus,
+  HCaptchaStatus,
+  setCognitoFlowStatus,
+  setHCaptchaStatus,
+  setUserChallengeDisbursed
+} from 'common/store/pages/audio-rewards/slice'
+import { setVisibility } from 'common/store/ui/modals/slice'
+import { increaseBalance } from 'common/store/wallet/slice'
+import { stringAudioToStringWei } from 'common/utils/wallet'
+import AudiusBackend from 'services/AudiusBackend'
+// Need the mock type to get the helper function that sets the config
+// eslint-disable-next-line jest/no-mocks-import
+import { MockRemoteConfigInstance } from 'services/remote-config/__mocks__/remote-config-instance'
+import { remoteConfigInstance } from 'services/remote-config/remote-config-instance'
+
+import rewardsSagas from './sagas'
+
+jest.mock('services/remote-config/remote-config-instance')
+jest.mock('utils/route/hashIds')
+jest.mock('services/AudiusBackend')
+function* saga() {
+  yield all(rewardsSagas().map(fork))
+}
+
+const testClaim: Claim = {
+  challengeId: 'connect-verified',
+  specifier: '1',
+  amount: 10
+}
+const testUser = {
+  user_id: 1,
+  handle: 'test_user',
+  wallet: 'test-wallet'
+}
+const testUserChallenge = {
+  challenge_id: 'connect-verified',
+  amount: 1,
+  is_complete: true
+}
+const claimAsyncProvisions: StaticProvider[] = [
+  [select(getAccountUser), testUser],
+  [
+    select.like({
+      selector: getUserChallenge,
+      args: [{ challengeId: testUserChallenge.challenge_id }]
+    }),
+    testUserChallenge
+  ]
+]
+
+const retryClaimProvisions: StaticProvider[] = [
+  [select(getClaimStatus), ClaimStatus.WAITING_FOR_RETRY],
+  [select(getClaimToRetry), testClaim]
+]
+
+const expectedRequestArgs = {
+  ...testClaim,
+  encodedUserId: undefined,
+  handle: 'test_user',
+  recipientEthAddress: 'test-wallet',
+  oracleEthAddress: 'oracle eth address',
+  quorumSize: 1,
+  endpoints: ['rewards attestation endpoints'],
+  AAOEndpoint: 'oracle endpoint'
+}
+beforeAll(() => {
+  ;(remoteConfigInstance as MockRemoteConfigInstance).__setConfig({
+    [IntKeys.ATTESTATION_QUORUM_SIZE]: 1,
+    [StringKeys.ORACLE_ETH_ADDRESS]: 'oracle eth address',
+    [StringKeys.ORACLE_ENDPOINT]: 'oracle endpoint',
+    [StringKeys.REWARDS_ATTESTATION_ENDPOINTS]: 'rewards attestation endpoints'
+  })
+  const oldConsoleError = console.error
+  jest.spyOn(console, 'error').mockImplementation((...args) => {
+    if (
+      args &&
+      args.length > 1 &&
+      args[1].toString().indexOf('User is blocked from claiming') < 0
+    ) {
+      oldConsoleError(args)
+    }
+  })
+})
+
+describe('Claim Rewards Async saga', () => {
```

Comment:
Previously our convention had been to name this file

store.test.js because it is testing a bit more than just the saga itself.
https://github.com/AudiusProject/audius-client/blob/master/src/store/lineup/store.test.js

thoughts? i think consistency is more important than semantics

Reply:
Sure, I have no preference here really.

Reply:
yeah i like this idea